.github/workflows/tests-python.yml

```diff
@@ -0,0 +1,76 @@
+name: Tests (Python)
+
+on:
+  push:
+    branches: [ main, ci-tests-native ]
+  pull_request:
+    branches: [ main, ci-tests-native ]
+
+  # Allow job to be triggered manually.
+  workflow_dispatch:
+
+# Cancel in-progress jobs when pushing to the same branch.
+concurrency:
+  cancel-in-progress: true
+  group: ${{ github.workflow }}-${{ github.ref }}
+
+jobs:
+  tests:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest"]
+        python-version: ["3.9", "3.10", "3.11"]
+        # In order to save resources, only run particular
+        # matrix slots on other OS than Linux.
+        include:
+          - os: "macos-latest"
+            python-version: "3.11"
+          #- os: "windows-latest"
+          #  python-version: "3.11"
```

Comment:
Let me know if you want to see any adjustments to the test matrix configuration.

I've not added Python 3.8 here, in order to save resources, and, in a similar spirit, the recipe will only run a single version of Python on macOS.

On regular operations, you would toggle to `fail-fast: true`, in order to save further resources: It will cancel other jobs when a test job fails on one test matrix slot, because it does not make sense to continue running them.

Using `fail-fast: false` makes sense when working on the workflow recipe itself, in order to get the big picture if specific tests would only fail on Windows, for example, and you don't want them to cancel the jobs running on Linux.

Reply:
With 1ff1f3666c, I've configured `fail-fast: true`, to save resources in the default case, where it does not make sense to run the full test matrix, because the output from a single instance will usually be enough to find out about the root cause of the error.